cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/common/StructuredRecordUtils.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.etl.common;
+
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Utils class that contains StructuredRecord related transformations.
+ */
+public class StructuredRecordUtils {
+
+  /**
+   * Converts the field names in the input {@link StructuredRecord} to a desired case
+   * @param input {@link StructuredRecord}
+   * @param fieldCase {@link FieldCase}
+   * @return {@link StructuredRecord} which contains field names confirming to the {@link FieldCase} passed in
+   * @throws Exception if there is a conflict in the field names while converting the case
+   */
+  public static StructuredRecord convertCase(StructuredRecord input, FieldCase fieldCase) throws Exception {
+    if (fieldCase.equals(FieldCase.NONE)) {
+      return input;
+    }
+
+    Schema oldSchema = input.getSchema();
+    Map<String, String> fieldNameMap = Maps.newHashMap();
+    List<Schema.Field> newFields = Lists.newArrayList();
+    for (Schema.Field field : oldSchema.getFields()) {
+      String newName = nameChange(field.getName(), fieldCase);
+      if (fieldNameMap.containsValue(newName)) {
+        // field name used already. indication of field names conflict. can't do anything.
+        throw new Exception(String.format(
```

Comment:
yeah, not sure if that will increase the size of this PR, but that sounds like a good exception. Is that exception supposed to be used elsewhere?